extensions/security/iprange/src/org/exist/security/realm/iprange/IPRangeServlet.java

```diff
@@ -63,12 +63,15 @@ protected void doPost(final HttpServletRequest request, final HttpServletRespons
 
 
         // Get reverse proxy header when available, otherwise use regular IP address
-        String ipAddress = request.getHeader("X-Forwarded-For");
-        if (ipAddress == null) {
+        String ipAddress = request.getHeader("X-Forwarded-For").replaceAll("\\s","");
+        // there may be a comma-separated chain of proxies
+        if (!ipAddress.isEmpty()) {
+            String[] xFFs = ipAddress.split(",");
+            if(xFFs.length > 1) ipAddress = xFFs[xFFs.length - 1];
```

Comment:
please could you describe which address you want to fetch? 1st? last?

Reply:
The last IP address should be used, because the values are only *added* when chained proxies are used. If no additional proxies are used, the XFF header only contains the remote address, which is the default.

Reply:
> In a chain of proxy servers where this is fully utilized, the first "for" parameter will disclose the client where the request was first made, followed by any subsequent proxy identifiers.  The last proxy in the chain is not part of the list of "for" parameters.  The last proxy's IP address, and optionally a port number, are, however, readily available as the remote IP address at the transport layer.  It can, however, be more relevant to read information about the last proxy from preceding "Forwarded" header field's "by" parameter, if present.

https://tools.ietf.org/html/rfc7239#page-6